includes/class-attachments.php

```diff
@@ -444,15 +451,31 @@ private static function save_attachment( $attachment_data, $post_id, $author_id
 			}
 		}
 
+		// Optimize images before sideloading (resize and convert to WebP).
+		$optimized = self::optimize_image( $tmp_file, self::MAX_IMAGE_DIMENSION );
+		if ( $optimized['changed'] ) {
+			$tmp_file = $optimized['path'];
+		}
+
 		// Prepare file array for WordPress.
+		$original_name = \basename( \wp_parse_url( $attachment_data['url'], PHP_URL_PATH ) );
+
+		// Update filename extension if format changed.
+		if ( $optimized['changed'] ) {
+			$new_ext       = \pathinfo( $tmp_file, PATHINFO_EXTENSION );
+			$original_name = \preg_replace( '/\.[^.]+$/', '.' . $new_ext, $original_name );
+		}
+
 		$file_array = array(
-			'name'     => \basename( \wp_parse_url( $attachment_data['url'], PHP_URL_PATH ) ),
+			'name'     => $original_name,
 			'tmp_name' => $tmp_file,
 		);
 
 		// Prepare attachment post data.
+		// Clear mime type if format changed so WordPress auto-detects it.
+		$mime_type = $optimized['changed'] ? '' : ( $attachment_data['mediaType'] ?? '' );
 		$post_data = array(
-			'post_mime_type' => $attachment_data['mediaType'] ?? '',
+			'post_mime_type' => $mime_type,
```

Comment:
Let's just remove this and let WP handle it? I've not checked how it works under the hood, this is just based on the comment in line 475.